src/functions/members.js

```diff
@@ -0,0 +1,49 @@
+import {db} from '../Firebase/FirebaseConfig';
+import { collection, getDocs, doc, getDoc, setDoc, addDoc, deleteDoc } from 'firebase/firestore';
+
+async function doesMemberExist(snapshot) {
+    if (!snapshot.exists()) throw new Error('Member not found.');
+}
+
+// Change 'Users' path to appropriate members path when the collection is created
+async function getMembers() {
+    const membersCollection = collection(db, 'users');
+    const membersSnapshot = await getDocs(membersCollection);
+    const membersList = membersSnapshot.docs.map(doc => doc.data());
+    return membersList;
+}
+
+async function getMemberById(id) {
+    const memberRef = doc(db, 'members', id);
+    const memberSnapshot = await getDoc(memberRef);
+    doesMemberExist(memberSnapshot);
+    return memberSnapshot.data();
+}
+
+async function addMember(member) {
+    if (!member.fullName || !member.email) {
+        throw new Error('Member must have a name and an email address.');
+    }
+    const newMemberRef = await addDoc(collection(db, 'members'), member);
+    return newMemberRef.id;
+}
+
+// The shouldMerge parameter is optional and defaults to false.
+// If set to true, the member object will be merged with the existing document and will not overwrite any existing fields.
```

Comment:
Not sure why you need to have this functionality/argument. Better off having separate functions for add and update.